ElastosDIDSDK/DID.swift

```diff
@@ -167,25 +217,38 @@ public class DID {
             }
         }
     }
+
+    /// Get all DID Documents from chain.
+    /// - Returns: return the handle to DID Document asynchronously.
+    @objc public func resolveHistoryAsync() -> AnyPromise {
+        return AnyPromise(__resolverBlock: { [self] resolver in
+            do {
+                resolver(try resolveHistory())
+            } catch let error  {
+                resolver(error)
+            }
+        })
+    }
 }
 
-extension DID: CustomStringConvertible {
+extension DID {
     func toString() -> String {
         return String("did:\(_method!):\(_methodSpecificId!)")
     }
 
     /// Get id string from DID.
-    public var description: String {
+    @objc public override var description: String {
         return toString()
     }
 }
 
-extension DID: Equatable {
-    func equalsTo(_ other: DID) -> Bool {
+extension DID {
```

Comment:
Be sure to remove the extension from Equatable. It could be wrong if DID is as a Key of map object.